File: resources/[qb]/qb-smallresources/client/tackle.lua
Citizen.CreateThread(function()
    while true do 
        if DGCore ~= nil then
            local ped = PlayerPedId()
            if not IsPedInAnyVehicle(ped, false) and GetEntitySpeed(ped) > 2.5 then
                if IsControlJustPressed(1, 19) then
                    Tackle()
                end
            else
                Citizen.Wait(250)
            end
        end

        Citizen.Wait(1)
    end
end)

RegisterNetEvent('tackle:client:GetTackled')
AddEventHandler('tackle:client:GetTackled', function()
	SetPedToRagdoll(PlayerPedId(), math.random(1000, 6000), math.random(1000, 6000), 0, 0, 0, 0) 
	TimerEnabled = true
	Citizen.Wait(1500)
	TimerEnabled = false
end)

function Tackle()
    closestPlayer, distance = DGCore.Functions.GetClosestPlayer()
    local closestPlayerPed = GetPlayerPed(closestPlayer)
    if(distance ~= -1 and distance < 2) then
        TriggerServerEvent("tackle:server:TacklePlayer", GetPlayerServerId(closestPlayer))
        TackleAnim()
    end
end

function TackleAnim()
    local ped = PlayerPedId()
    if not DGCore.Functions.GetPlayerData().metadata["ishandcuffed"] and not IsPedRagdoll(ped) then
        RequestAnimDict("swimming@first_person@diving")
        while not HasAnimDictLoaded("swimming@first_person@diving") do
            Citizen.Wait(1)
        end
        if IsEntityPlayingAnim(ped, "swimming@first_person@diving", "dive_run_fwd_-45_loop", 3) then
            ClearPedTasksImmediately(ped)
        else
            TaskPlayAnim(ped, "swimming@first_person@diving", "dive_run_fwd_-45_loop" ,3.0, 3.0, -1, 49, 0, false, false, false)
            Citizen.Wait(250)
            ClearPedTasksImmediately(ped)
            SetPedToRagdoll(ped, 150, 150, 0, 0, 0, 0)
        end
    end
end

File: resources/[dg]/dg-misc/typescript/src/client/services/tackle.ts
import { Events, Keys, Police, Util, Weapons } from '@dgx/client';

Events.onNet('misc:tackle:do', () => {
  SetPedToRagdoll(
    PlayerPedId(),
    Util.getRndInteger(3000, 6000),
    Util.getRndInteger(3000, 6000),
    0,
    false,
    false,
    false
  );
});

const tackleInit = async () => {
  if (!canTackle()) return;

  const closestPlayer = Util.getClosestPlayerInDistanceAndOutsideVehicle(2);
  if (!closestPlayer) return;

  Events.emitNet('misc:tackle:server', GetPlayerServerId(closestPlayer));

  const ped = PlayerPedId();
  await Util.loadAnimDict('swimming@first_person@diving');

  if (IsEntityPlayingAnim(ped, 'swimming@first_person@diving', 'dive_run_fwd_-45_loop', 3)) {
    ClearPedTasksImmediately(ped);
  } else {
    TaskPlayAnim(
      ped,
      'swimming@first_person@diving',
      'dive_run_fwd_-45_loop',
      3.0,
      3.0,
      -1,
      49,
      0,
      false,
      false,
      false
    );
    await Util.Delay(250);
    ClearPedTasksImmediately(ped);
    SetPedToRagdoll(ped, 150, 150, 0, false, false, false);
  }
};

Keys.onPressDown('tackle', () => {
  tackleInit();
});

Keys.register('tackle', '(gov) iemand tacklen', 'LMENU');

const canTackle = () => {
  const ped = PlayerPedId();
  return (
    !IsPedInAnyVehicle(ped, true) &&
    GetEntitySpeed(ped) > 2.5 &&
    IsPedRunning(ped) &&
    !IsPedRagdoll(ped) &&
    !Police.isCuffed() &&
    !Police.isEscorting() &&
    IsPedJumping(ped) &&
    Weapons.getCurrentWeaponData() === null &&
    !IsPedClimbing(ped) &&
    !IsPedSwimming(ped) &&
    !IsPedSwimmingUnderWater(ped)
  );
};
